File: flutter-app/lib/models/recipe.dart
class Recipe {
  final int id;
  final String title;
  final String image;
  final int? readyInMinutes;
  final int? servings;
  final double? spoonacularScore;
  final List<Ingredient> extendedIngredients;
  final List<Ingredient> usedIngredients;
  final List<Ingredient> missedIngredients;
  final List<AnalyzedInstruction> analyzedInstructions;
  final String? summary;
  final String? cuisines;
  final String? dishTypes;

  Recipe({
    required this.id,
    required this.title,
    required this.image,
    this.readyInMinutes,
    this.servings,
    this.spoonacularScore,
    this.extendedIngredients = const [],
    this.usedIngredients = const [],
    this.missedIngredients = const [],
    this.analyzedInstructions = const [],
    this.summary,
    this.cuisines,
    this.dishTypes,
  });

  factory Recipe.fromJson(Map<String, dynamic> j) => Recipe(
        id: j['id'],
        title: j['title'] ?? '',
        image: j['image'] ?? '',
        readyInMinutes: j['readyInMinutes'],
        servings: j['servings'],
        spoonacularScore: (j['spoonacularScore'] as num?)?.toDouble(),
        extendedIngredients: (j['extendedIngredients'] as List? ?? [])
            .map((e) => Ingredient.fromJson(e))
            .toList(),
        usedIngredients: (j['usedIngredients'] as List? ?? [])
            .map((e) => Ingredient.fromJson(e))
            .toList(),
        missedIngredients: (j['missedIngredients'] as List? ?? [])
            .map((e) => Ingredient.fromJson(e))
            .toList(),
        analyzedInstructions: (j['analyzedInstructions'] as List? ?? [])
            .map((e) => AnalyzedInstruction.fromJson(e))
            .toList(),
        summary: j['summary'],
      );
}

class Ingredient {
  final String original;
  final String name;
  final double amount;
  final String unit;

  Ingredient({required this.original, required this.name, required this.amount, required this.unit});

  factory Ingredient.fromJson(Map<String, dynamic> j) => Ingredient(
        original: j['original'] ?? '',
        name: j['name'] ?? '',
        amount: (j['amount'] as num?)?.toDouble() ?? 0,
        unit: j['unit'] ?? '',
      );
}

class AnalyzedInstruction {
  final List<Step> steps;
  AnalyzedInstruction({required this.steps});
  factory AnalyzedInstruction.fromJson(Map<String, dynamic> j) => AnalyzedInstruction(
        steps: (j['steps'] as List? ?? []).map((e) => Step.fromJson(e)).toList(),
      );
}

class Step {
  final int number;
  final String step;
  Step({required this.number, required this.step});
  factory Step.fromJson(Map<String, dynamic> j) =>
      Step(number: j['number'] ?? 0, step: j['step'] ?? '');
}

File: android-app/app/src/main/java/com/flavorfind/app/domain/model/Recipe.kt
package com.flavorfind.app.domain.model

data class Recipe(
    val id: Int,
    val title: String,
    val image: String,
    val readyInMinutes: Int?,
    val servings: Int?,
    val spoonacularScore: Double?,
    val summary: String?,
    val cuisines: List<String> = emptyList(),
    val dishTypes: List<String> = emptyList(),
    val extendedIngredients: List<Ingredient> = emptyList(),
    val usedIngredients: List<Ingredient> = emptyList(),
    val missedIngredients: List<Ingredient> = emptyList(),
    val analyzedInstructions: List<AnalyzedInstruction> = emptyList(),
)

data class Ingredient(
    val id: Int = 0,
    val name: String,
    val original: String,
    val amount: Double,
    val unit: String,
    val image: String = "",
)

data class AnalyzedInstruction(
    val name: String = "",
    val steps: List<Step> = emptyList(),
)

data class Step(
    val number: Int,
    val step: String,
    val ingredients: List<StepIngredient> = emptyList(),
)

data class StepIngredient(
    val id: Int,
    val name: String,
    val image: String = "",
)
